packages/mobile/src/transactions/feed/queryHelper.ts

```diff
@@ -0,0 +1,269 @@
+import { useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { useDispatch, useSelector } from 'react-redux'
+import { showError, showMessage } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import config from 'src/geth/networkConfig'
+import useInterval from 'src/hooks/useInterval'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import { updateTransactions } from 'src/transactions/actions'
+import { TokenTransaction } from 'src/transactions/types'
+import Logger from 'src/utils/Logger'
+import { walletAddressSelector } from 'src/web3/selectors'
+
+export interface QueryHookResult {
+  loading: boolean
+  error: Error | undefined
+  transactions: TokenTransaction[]
+  fetchingMoreTransactions: boolean
+  fetchMoreTransactions: () => void
+}
+interface PageInfo {
+  startCursor: string
+  endCursor: string
+  hasNextPage: boolean
+  hasPreviousPage: boolean
+}
+export interface QueryResponse {
+  data: {
+    tokenTransactionsV2: {
+      transactions: TokenTransaction[]
+      pageInfo: PageInfo
+    }
+  }
+}
+
+const TAG = 'transactions/feed/queryHelper'
+
+// Query poll interval
+const POLL_INTERVAL = 10000 // 10 secs
+
+function addDeduplicatedTransactions(
+  fetchedTransactions: TokenTransaction[],
+  setFetchedTransactions: (txs: TokenTransaction[]) => void
+) {
+  return (transactions: TokenTransaction[]) => {
+    const currentHashes = new Set(fetchedTransactions.map((tx) => tx.transactionHash))
+
+    const transactionsWithoutDuplicatedHash = fetchedTransactions.concat(
+      transactions.filter((tx) => !currentHashes.has(tx.transactionHash))
+    )
+
+    transactionsWithoutDuplicatedHash.sort((a, b) => {
+      return b.timestamp - a.timestamp
+    })
+
+    setFetchedTransactions(transactionsWithoutDuplicatedHash)
+  }
+}
+
+export function useFetchTransactions(): QueryHookResult {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const address = useSelector(walletAddressSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+
+  const [pageInfo, setPageInfo] = useState<PageInfo | null>(null)
+  const [fetchingMoreTransactions, setFetchingMoreTransactions] = useState(false)
+  const [transactions, setTransactions] = useState<TokenTransaction[]>([])
```

Comment:
i had a thought, let me know what you think:

i think `transactions` and `setTransactions` and `addDeduplicatedTransactions` can be abstracted to a hook? the hook could look something like ` const [transactions, setTransactions] = useDeduplicatedTransactions()`, where the hook can manage the `transactions` state variable + a "setTransactions" function for handling logic for deduplicating the transactions. the benefit would be that this `useFetchTransactions` hook has one less state variable to manage (and also doesn't need to pass the outputs of this state hook into another function)

this is a big of code golf, what you have here is fine too, so i'm curious for your thoughts!